Add an article about using "containers" such as docker

chrisrichardson opened the pull request:
This is up for comments. I can add more, or change the style - let me know any feedback.

maherou commented:
Chris,

A few comments:

When I ran the first example command:

docker run -ti ubuntu

I had the following issues:
- My VPN would not allow me to get docker content remotely (I mention this because it could be a showstopper for some).
- After suspending my VPN, I was able to execute the comment, but it took time to download content (again, just something to mention, perhaps).
- On the third try it worked very well.


Second, I tried the next example you give in the text, creating a Dockerfile with this content:

**s1000631:docker-test maherou$ more Dockerfile 
FROM ubuntu:16.10

RUN apt-get update && apt-get install -y python-numpy python-scipy python-matplotlib**

I ran the following command in the same directory as the Dockerfile, and got the following output.

**s1000631:docker-test maherou$ docker build --tag mypythonimage
"docker build" requires exactly 1 argument(s).
See 'docker build --help'.

Usage:  docker build [OPTIONS] PATH | URL | -

Build an image from a Dockerfile**

I was not able to get past this error using your instructions.  Can you figure out how to address this?

Thanks.

chrisrichardson commented:
Thanks for testing, Mike. I've added a line about VPN and DNS (which also sometimes causes problems). Also corrected my typo for the "docker build" command.

rinkug commented:
Published at : https://bssw.io/items/exploring-containers-for-research-software
